libs/platform/src/lib/table/table.component.ts

```diff
@@ -539,6 +539,19 @@ export class TableComponent<T = any>
         );
     }
 
+    /**
+     *  Whether the selection column should be shown,
+     *  If there are no visible rows or visible columns
+     *  or selection mode is none, then selection column should not be shown.
+     **/
+    get isSelectionColumnShown(): boolean {
```

Comment:
Personally I'd rather use a function call in the template than a getter because calling a getter in a template appears to be simply accessing a value, but is actually calling a function anyways, so it'll be harder to catch as a potential performance issue. Leaving a function call in a template will be easily to identify for any performance issues in the future

Reply:
This thing is OnPush, this function will be called only on change detection cycles

Reply:
I get your point, but I do not see function benefiting us at the moment. With the ng16, signal instead of this will be on point

Reply:
**Moved to the function**